x/model/types/messages_model_version_test.go

```diff
@@ -1,74 +1,736 @@
+				msg.OtaFileSize = 0
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaFileSize > 0 when OtaUrl is set",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaUrl = "https://sampleflowurl.dclmodel"
+				msg.OtaFileSize = 1
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksum is omitted when OtaUrl is omitted",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaUrl = ""
+				msg.OtaChecksum = ""
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksum is set when OtaUrl is set",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaUrl = "https://sampleflowurl.dclmodel"
+				msg.OtaChecksum = tmrand.Str(32)
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksum length == 64",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaChecksum = tmrand.Str(64)
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksumType == 0 when OtaUrl is omitted",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaUrl = ""
+				msg.OtaChecksumType = 0
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksumType != 0 when OtaUrl is set",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaUrl = "https://sampleflowurl.dclmodel"
+				msg.OtaChecksumType = 1
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksumType == 1",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaChecksumType = 1
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "OtaChecksumType == 65535",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.OtaChecksumType = 65535
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion < MaxApplicableSoftwareVersion",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.MinApplicableSoftwareVersion = 5
+				msg.MaxApplicableSoftwareVersion = 10
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "MinApplicableSoftwareVersion == MaxApplicableSoftwareVersion",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.MinApplicableSoftwareVersion = 7
+				msg.MaxApplicableSoftwareVersion = 7
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "ReleaseNotesUrl is omitted",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.ReleaseNotesUrl = ""
+				return msg
+			}(validMsgCreateModelVersion()),
+		},
+		{
+			name: "ReleaseNotesUrl length == 256",
+			msg: func(msg *MsgCreateModelVersion) *MsgCreateModelVersion {
+				msg.ReleaseNotesUrl = "https://sampleflowurl.dclauth/" + tmrand.Str(256-30) // length = 256
+				return msg
+			}(validMsgCreateModelVersion()),
```

Comment:
Please add the following positive cases:
- SoftwareVersion=0
- MinApplicableSoftwareVersion=0 and MaxApplicableSoftwareVersion=0
- MinApplicableSoftwareVersion=0 and MaxApplicableSoftwareVersion>0

Reply:
Done.